FifthLesson/VectorProgram/VectorProgram/ProgramRun.cs

```diff
@@ -13,87 +9,125 @@ class ProgramRun
+            Console.WriteLine(string.Empty);
         }
 
-        private void WriteErrorMessage()
+        private void ShowVectorMultiplication(Vector first, Vector second)
         {
-            Console.WriteLine("You entered non-correct numbers.");
+            var vectorMultiplication = Vector.CalculateVectorMultiplicate(first, second);
+            Console.WriteLine($"VectorMultiplication: {first} x {second} = {VectorMultiplication}");
+            Console.WriteLine(string.Empty);
+        }
+
+        private void ShowAngleBetweenVectors(Vector vectorFirst, Vector vectorSecond)
+        {
+            var angle = vectorFirst.CalculateAngleBetweenVectors(vectorSecond);
+            Console.WriteLine($"The angle between {vectorFirst} and {vectorSecond} = {Math.Abs(angle)}");
+            Console.WriteLine(string.Empty);
+        }
+
+        private void ShowMultiplicationVectorsByNumber(Vector first, Vector second)
+        {
+            var numericMultiplier = GetUserNumericMultiplier();
+            Console.WriteLine("Multiplication of vectors by number:");
+
+            Console.WriteLine($"{first} * {numericMultiplier} = {(first * numericMultiplier)}");
+            Console.WriteLine($"{second} * {numericMultiplier} = {(second * numericMultiplier)}");
+            Console.WriteLine(string.Empty);
+            Console.WriteLine($"{numericMultiplier} * {first} = {(numericMultiplier * first)}");
+            Console.WriteLine($"{numericMultiplier} * {second} = {(numericMultiplier * second)}");
         }
         
+        private Vector GetUserVector(string orderByVectors)
+        {
+            var userInput = RequestUserInput(DataType.Vector, $"Enter the coordinates of the {orderByVectors} three-dimensional vector separated by a space:");
+            return ParseUserInputAndCreateVector(userInput);
+        }
+
+        private double GetUserNumericMultiplier()
+        {
+            double numericInput;
+            var userInput = RequestUserInput(DataType.Multiplier, "Enter a number to produce a vector by a number:");
+
+            double.TryParse(userInput.Replace(",", "."), NumberStyles.Number, CultureInfo.CreateSpecificCulture("en-US"), out numericInput);
+            return numericInput;
+        }
+                
         private Vector ParseUserInputAndCreateVector(string userInput)
         {
-            var coords = VerificateInputData.ParseUserInput(userInput);
+            var coords = DataParser.ParseStringToArray(userInput);
             return new Vector(coords[0], coords[1], coords[2]);
         }
-
-        private string ReadInputCoordsAndVerificate()
+        
+        private string RequestUserInput(DataType dataType, string welcomMessage)
         {
-            var userInput = Console.ReadLine();
-            while(!VerificateInputData.VerifyInputDataCoordsVector(userInput))
+            var userInput = string.Empty;
+            var correctInput = false;
```

Comment:
Rename to **isUserInputCorrect** or smth else, because it is boolean variable and we are using it as a flag

Reply:
Corrected.